### PR TITLE
Add dockerfile for container build

### PR DESCRIPTION
**Pull request by mgazza**
Add the ability to build the [tutorial](resources/tutorial_building) using docker
The Dockerfile builds a container which runs build.sh.
The build files can then be extracted using `docker cp`
see [readme](./readme.md#docker)

@gabonator I would appreciate some help building some of the apps within this container :)

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -x
+mkdir output
+
+cd ./system/os_host
+./build.sh
+cp ./build/system_la104.hex ../../output/
+cd ../os_library
```
Comment (gabonator) [inline build.sh:9]: the correct order for building is:
1. os_library - not only it builds dummy library for applications, but also this updates the list of exported methods for host (in Shared.h).
2. os_host

And I suggest building following applications:
1. apps_shell/test29_fileman - file manager 
2. LA104: apps_featured/test49_gpio - GPIO control/pwm generator
3. LA104: apps_featured/test63_ws2812 - nice application for controlling addressable RGB leds
4. LA104: apps_featured/test33_temper - digital thermometer using DS1820
5. LA104/DS213/DS203: apps_ds213/61_devinfo - device info
5. DS203: apps_featured/69_webusbosc - web usb oscilloscope
6. DS203: apps_featured/55_gabosc - oscilloscope

Comment (gabonator) [reply build.sh:9]: I also suggest removing the hardcoded arm toolchain paths from build scripts at least from those folders referenced by this docker file.

Comment (mgazza) [reply build.sh:9]: I think i'm going to get some build errors. I get quite a few from compiling the buildall.sh. I will post them in here and if you could help me getting them building id appreciate that. 
I've also had chance to flash the checked in /system/release binary's theres quite a few apps that don't appear to work or crash. Would you like me to open issues for each?

Comment (gabonator) [reply build.sh]: Here is what I get after running buildall.sh:
```
Building host...
20003000 00000000 D _addressRamBegin
200042cc 00000000 B _addressRamEnd
0800c000 00000000 T _addressRomBegin
080126f8 00000000 T _addressRomEnd
20000000 00000000 D _addressRamBegin
20002764 00000000 B _addressRamEnd
08008000 00000000 T _addressRomBegin
08012bf0 00000000 T _addressRomEnd
20000000 00000000 D _addressRamBegin
200025e8 00000000 B _addressRamEnd
08008000 00000000 T _addressRomBegin
08013060 00000000 T _addressRomEnd
Building library
apps_featured/69_webusbosc/: Success!
apps_featured/74_invtvisual/: Success!
apps_featured/80_rftool/: No build script
apps_featured/test21_midiplay/: Success!
apps_featured/test33_temper/: Success!
apps_featured/test49_gpio/: Success!
apps_featured/test63_ws2812/: Success!
apps_shell/test28_shell/: Success!
apps_shell/test29_fileman/: Success!
apps_experiments/72_uartmon/: Success!
apps_experiments/73_invt/: Success!
apps_experiments/75_invtemu/: Success!
apps_experiments/76_cc1101send/: Success!
apps_experiments/77_dmatest/: Success!
apps_experiments/78_uartgen/: Success!
apps_experiments/79_cc1101osc/: No build script
apps_experiments/81_image/: Success!
apps_experiments/82_specan/: Build failed        
apps_experiments/test10_i2c_direct/: Success!
apps_experiments/test12_uart_api/: Success!
apps_experiments/test16_cc1101/: Success!
apps_experiments/test1_noimport/: Success!
apps_experiments/test20_uartplay/: Success!
apps_experiments/test27_irsend/: Success!
apps_experiments/test2_import/: Success!
apps_experiments/test32_graph/: Success!
apps_experiments/test38_cc1101/: Success!
apps_experiments/test39_lcd/: Success!
apps_experiments/test3_gui/: Success!
apps_experiments/test40_fpga/: Success!
apps_experiments/test42_ccview/: Success!
apps_experiments/test43_dumpmem/: Success!
apps_experiments/test44_hardfault/: Success!
apps_experiments/test46_console/: Success!
apps_experiments/test48_scroll/: Success!
apps_experiments/test56_ds1307/: Success!
apps_experiments/test57_ds3231/: Success!
apps_experiments/test5_blink_hex/: Success!
apps_experiments/test68_webusb/: Success!
apps_experiments/test6_pwm_hex/: Success!
apps_experiments/test8_ds_app/: Success!
apps_experiments/test9_dht_app/: Success!
apps/83_debugging/: Success!
apps/test13_mp3/: Success!
apps/test15_charmap/: Success!
apps/test17_official/: Success!
apps/test22_sequencer/: Success!
apps/test23_screenshot/: Success!
apps/test30_dcf77/: Success!
apps/test31_testsignal/: Success!
apps/test34_scope/: Success!
apps/test35_view/: Success!
apps/test36_esp_server/: Success!
apps/test37_i2cscan/: Success!
apps/test41_alchemy/: Success!
apps/test45_simcom/: Success!
apps/test4_snake/: Success!
apps/test70_webusb/: Success!
apps/test7_pwm_app/: Success!
apps_arduino/test11_i2c_api/: Success!
apps_arduino/test14_apds9960/: Success!
apps_arduino/test18_onewire/: Success!
apps_usb/test24_usbcdc/: Success!
apps_usb/test25_hid/: Success!
apps_usb/test26_midi/: Success!
apps_usb/test32_sump/: Success!
apps_usb/test64_remote/: Success!
apps_usb/test67_webusb/: Success!
mkdir: release: File exists
mkdir: release/bin: File exists
cp: apps_ds213/58_gabmini//build/*.elf: No such file or directory
cp: apps_ds213/61_devinfo//build/*.elf: No such file or directory
cp: apps_ds213/71_dmastreamer//build/*.elf: No such file or directory
cp: apps_experiments/79_cc1101osc//build/*.elf: No such file or directory
cp: apps_experiments/82_specan//build/*.elf: No such file or directory
cp: apps_featured/80_rftool//build/*.elf: No such file or directory
cp: apps_unfinished/midi//build/*.elf: No such file or directory
cp: apps_unfinished/test32_sump//build/*.elf: No such file or directory
cp: apps_unfinished/test47_eeprom//build/*.elf: No such file or directory
```

ufff, opening case for each app is not the best idea. Many of them are just some useless experiments. But if you want, you can make single case with a list of non working apps...

**Pull request merged**